### PR TITLE
Upgrade to pgTAP 1.3.1

### PR DESCRIPTION
**Pull request by halostatue**
Planned for release as 2.5.0.

- Upgraded pgTAP to 1.3.1.

- Update support for the released version of PostgreSQL 16.

- Changed maintenance procedures to:

  - Add the version of Alpine to `package-versions.json` and use that for image
    builds.
  - Change maintenance procedure to automatically generate the
    `build/pgtap/Dockerfile`. The latest versions required some updates because
    of CLANG and LLVM version mismatches.

Signed-off-by: Austin Ziegler <aziegler@kineticcommerce.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,50 +1,68 @@
...
-RUN chmod +x /home/sqitch/bin/* \
-    && chown -R sqitch:sqitch /home
+
+RUN <<FINALIZE
+set -eux
+
+adduser --disabled-password sqitch
+mkdir -p /opt /home/sqitch/bin
+
+echo $__DOCKERFILE_VERSION__ > /home/sqitch/VERSION
```
Comment (mikestok) [inline Dockerfile:61]: (nit, no change required) The name of /home/sqitch/VERSION suggests to me that it's to do with the version of sqitch. Maybe call the file /home/sqitch/DOCKERFILE_VERSION?

**Review by mikestok — APPROVED** [review]
It looks OK to me, but it's big and my attention flagged.

I skipped a lot of the pgtap stuff, and I'm not sure if my time spend reading the build stuff is that well spent.

"Nothing seems obviously wrong" is a lower quality bar than I'd like to use.

**Comment by halostatue** [comment]
> It looks OK to me, but it's big and my attention flagged.
> 
> I skipped a lot of the pgtap stuff

That's effectively generated.

> , and I'm not sure if my time spend reading the build stuff is that well spent.

This is the main change that was made because building with pgTAP 1.3.1 failed. Took too long to figure out.

File: Dockerfile (hunk 1 above)
Comment (halostatue) [reply Dockerfile:61]: I might change the username from `sqitch` to `sqitcher` or something like that in the future to reflect that this is the user for the image in general. `VERSION` is a typical name for this.

**Pull request merged**